lab_3_generate_by_ngrams/main.py

```diff
@@ -104,6 +159,18 @@ def _put(self, element: str) -> None:
         In case of corrupt input arguments or invalid argument length,
         an element is not added to storage
         """
+        if not isinstance(element, str) or len(element) != 1:
+            return None
+
+        if element in self._storage:
+            return self._storage[element]
+        key = len(self._storage)
+        while key <= len(self._storage):
```

Comment:
1. This `while` could easily be `for`
2. Given your code, is it likely that there will be an unused identifier smaller than the length of the storage?